crates/bevy_ecs/src/world/mod.rs

```diff
@@ -889,8 +889,10 @@ impl World {
     #[inline]
     pub fn get_resource<R: Resource>(&self) -> Option<&R> {
         let component_id = self.components.get_resource_id(TypeId::of::<R>())?;
+        let p = self.get_resource_by_id(component_id)?;
```

Comment:
As in #6304, this adds an extra check for thread safety that isn't needed.

Reply:
How ought it be modified?

Reply:
We could either:
1. eat the performance cost
2. duplicate the code
3. quasi-duplicate the code using something like const generics
4. create an internal unsafe method that assumes the checks has been done if needed and then call that inside of get_resource_by_id (and directly here)
5. leave the method duplication raised in #6294 alone

I think 4 is my preference.

Reply:
4 basically recreates `get_resource_with_id` as is. The code contained within also isn't too complex. I'd rather us just duplicate the code in the few places it's used right now.

Reply:
Yeah, maybe the right fix is just docs...

Reply:
Given these comments I will just withdraw this PR.  Agreed?